e2e/cypress/integration/files_and_attachments/upload_files_spec.js

```diff
@@ -179,20 +189,134 @@ describe('Upload Files', () => {
+
+            // # Get the last post
+            cy.getLastPostId().then((lastPostId) => {
+            // # Scan inside of the last post message
+                cy.get(`#${lastPostId}_message`).and('be.visible').within(() => {
+                // * Check if the attached file is in the post and then click to open preview
+                    cy.findByLabelText(`file thumbnail ${file}`).should('be.visible').click();
+                });
+            });
+
+            // * Verify preview modal is opened
+            cy.get('.a11y__modal').should('exist').and('be.visible').
+                within(() => {
+                    // * Check if get public link button is present and click it
+                    cy.findByText('Get a public link').should('be.visible').click({force: true});
+                });
+
+            // # Wait a little for url to be generated
+            cy.wait(TIMEOUTS.ONE_SEC);
+
+            // * Verify copy public link modal is opened
+            cy.get('.a11y__modal.modal-dialog').should('exist').and('be.visible').
+                within(() => {
+                    // * Verify that copy link button is present
+                    cy.findByText('Copy Link').should('be.visible');
+
+                    // # Get the copy link of the attachment and save for later purpose
+                    cy.get('#linkModalTextArea').invoke('text').as(`publicLinkOfAttachment-${file}`);
+
+                    cy.get('.modal-footer').should('exist').within(() => {
+                        // # Click close modal
+                        cy.findByText('Close').should('be.visible').click();
+                    });
+                });
+
+            cy.get(`@publicLinkOfAttachment-${file}`).then((publicLinkOfAttachment) => {
+                // # Post the link of attachment as a message
+                cy.postMessageQuickly(publicLinkOfAttachment);
+
+                // * Check the attachment url contains the attachment
+                downloadAttachmentAndVerifyItsProperties(publicLinkOfAttachment, file, 'inline');
+            });
+        });
+    });
+});
 
+function downloadAttachmentAndVerifyItsProperties(fileURL, filename, httpContext) {
     // * Verify it has not empty download link
-    cy.request(fileAttachment.attr('href')).then((response) => {
+    cy.request(fileURL).then((response) => {
         // * Verify that link can be downloaded
         expect(response.status).to.equal(200);
 
-        // * Verify if link is an attachment that can be saved locally
+        // * Verify if link is an appropriate httpContext for opening in new tab or same and that can be saved locally
         // and it contains the correct filename* which will be used to name the downloaded file
         expect(response.headers['content-disposition']).to.
-            equal(`attachment;filename="${filename}"; filename*=UTF-8''${filename}`);
+            equal(`${httpContext};filename="${filename}"; filename*=UTF-8''${filename}`);
```

Comment:
Could you confirm if all are passing on you? I've got failing test here.  See screenshot.
<img width="684" alt="Screen Shot 2020-09-01 at 6 34 20 PM" src="https://user-images.githubusercontent.com/5334504/91839597-e8ddae80-ec81-11ea-9104-5cdf59108325.png">